poolakey/src/main/java/ir/cafebazaar/poolakey/billing/connection/ReceiverBillingConnection.kt

```diff
@@ -0,0 +1,411 @@
+                consumeFailed.invoke(ConsumeFailedException())
+            }
+        }
+    }
+
+    private fun onBillingSupportActionReceived(extras: Bundle?) {
+        val isResponseSucceed = isResponseSucceed(extras)
+        val isSubscriptionSupport = isSubscriptionSupport(extras)
+
+        when {
+            isResponseSucceed && isSubscriptionSupport -> {
+                connectionCallbackReference?.get()?.connectionSucceed?.invoke()
+            }
+            !isResponseSucceed -> {
+                connectionCallbackReference?.get()?.connectionFailed?.invoke(
+                    IAPNotSupportedException()
+                )
+            }
+            else -> {
+                connectionCallbackReference?.get()?.connectionFailed?.invoke(
+                    SubsNotSupportedException()
+                )
+            }
+        }
+    }
+
+    private fun isSubscriptionSupport(extras: Bundle?): Boolean {
+        val isSubscriptionSupport = extras?.getBoolean(KEY_SUBSCRIPTION_SUPPORT) ?: false
+        return !paymentConfiguration.shouldSupportSubscription || isSubscriptionSupport
+    }
+
+    private fun isResponseSucceed(extras: Bundle?): Boolean {
+        return extras?.getInt(RESPONSE_CODE) == BazaarIntent.RESPONSE_RESULT_OK
+    }
+
+    private fun isBundleSignatureValid(extras: Bundle?): Boolean {
+        return getSecureSignature() == extras?.getString(KEY_SECURE)
+    }
+
+    private fun registerBroadcast() {
+        BillingReceiver.addObserver(requireNotNull(receiverCommunicator))
+    }
+
+    private fun getNewIntentForBroadcast(): Intent {
+        val bundle = Bundle().apply {
+            putString(KEY_PACKAGE_NAME, contextReference?.get()?.packageName)
+            putString(KEY_SECURE, getSecureSignature())
+            putInt(KEY_API_VERSION, Billing.IN_APP_BILLING_VERSION)
+        }
+        return Intent().apply {
+            `package` = BAZAAR_PACKAGE_NAME
+            putExtras(bundle)
+        }
+    }
+
+    private fun getSecureSignature(): String {
+        val rsaKey = (paymentConfiguration.localSecurityCheck as? SecurityCheck.Enable)
+            ?.rsaPublicKey
+
+        return rsaKey ?: DEFAULT_SECURE_SIGNATURE
+    }
+
+    private fun sendBroadcast(intent: Intent) {
+        contextReference?.get()?.sendBroadcast(intent)
+    }
+
+    companion object {
+        private const val BAZAAR_WITH_RECEIVER_CONNECTION_VERSION = 801301
+
+        private const val DEFAULT_SECURE_SIGNATURE = "secureBroadcastKey"
+        private const val ACTION_BAZAAR_BASE = "com.farsitel.bazaar."
+        private const val ACTION_BAZAAR_POST = ".iab"
+
+        private const val ACTION_BILLING_SUPPORT = ACTION_BAZAAR_BASE + "billingSupport"
+        private const val ACTION_CONSUME: String = ACTION_BAZAAR_BASE + "consume"
+        private const val ACTION_PURCHASE: String = ACTION_BAZAAR_BASE + "purchase"
+        private const val ACTION_QUERY_PURCHASES: String = ACTION_BAZAAR_BASE + "getPurchase"
```

Comment:
I think you can remove the type definition.

Reply:
Removed